test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -0,0 +1,203 @@
+from core.test_run import TestRun
+from test_tools.dd import Dd
+from test_utils.size import Size, Unit
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_remove_multilevel_cas(cache_mode):
+    """
+        title: Test if OpenCAS not allow remove the core on 1 level cache when it's used by 2 level.
+        description: |
+          Negative test of the ability to remove core used by nested OpenCAS.
+        pass_criteria:
+          - No system crash.
+          - Cache do not allow removing the core on 1st level cache when its used by level 2nd.
+    """
+    with TestRun.step("Prepare two cache and one core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(1, Unit.GibiByte), Size(1, Unit.GibiByte)])
+        cache_dev1 = cache_dev.partitions[0]
+        cache_dev2 = cache_dev.partitions[1]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start first cache instance"):
+        cache1 = casadm.start_cache(cache_dev1, cache_mode, force=True)
+
+    with TestRun.step("Add core device to first cache instance."):
+        core1 = cache1.add_core(core_dev)
+
+    with TestRun.step("Start second cache instance"):
+        cache2 = casadm.start_cache(cache_dev2, cache_mode, force=True)
+
+    with TestRun.step("Add first cache's exported object as a core to second cache instance."):
+        cache2.add_core(core1)
+
+    with TestRun.step("Try to remove core from 1st level cache."):
+        output = TestRun.executor.run_expect_fail(cli.remove_core_cmd(cache_id=str(cache1.cache_id),
+                                                                      core_id=str(core1.core_id),
+                                                                      force=True))
+        if not output.stderr:
+            TestRun.fail("Core remove succeeded (should fail)!")
+
+    with TestRun.step("Stop all caches."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", [CacheMode.WB, CacheMode.WA])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_one_core_remove(cache_mode):
+    """
+        title: Test if OpenCAS correctly handles the remove of one of the core devices.
+        description: |
+          When one core device is removed from a cache instance all blocks previously occupied
+          by data from that core device should be removed. That means that number of free
+          cache blocks should increase by number of removed blocks.
+        pass_criteria:
+          - No system crash.
+          - Second core is able to use OpenCAS.
```

Comment:
add checking free blocks after core removal to pass criteria and to the test. add checking blocks read from cache before and after reading from the remaining exported object